### ISSUE TITLE
Reload server config

### ISSUE THREAD

**Issue by remram44** [mumble-voip/mumble#1680]
There seem to be no way (that I could find in docs, anyway) to reload the config file or database without restarting the server. It is pretty common for servers to reload their setting upon receiving some signal, like SIGHUP.

**Comment by mkrautz** [comment]
Could you elaborate on which murmur.ini options you would like to change and have the server reload them?

In the general case, a LOT of the options in murmur.ini are simply defaults for virtual servers.  This means that virtual servers will only use them if they don't have the corresponding option set.

Virtual servers are managed via RPC -- and practically all options can be set on a per-server basis. Most other options I can think of wouldn't really work in a hot reload situation and would require a full server restart anyway.

So, to sum up: Murmur is optimized for hosts that manage their server fleet via RPC. Pretty much everything can be done via RPC -- though some of those will require a restart a particular virtual server to take effect.

If you have any specific murmur.ini options in mind that would work great for hot-reloading, it's probably something we would consider.

Thanks for the report!

**Comment by remram44** [comment]
I'm a lot more interested in changes in the sqlite database than then INI file. I was surprised to see that a change in ACL directly in the database didn't reflect on the server before a restart.

**Comment by mkrautz** [comment]
The database is considered private, and not part of any public interface for interacting with Murmur.

While it probably would be technically possible to obtain consistency while allowing users to write to the database, it would complicate the database layer considerably. Right now, it's just dumb, structured storage.

Were we to allow it, we would need to be informed of every change made to the database, as it happens, and sync our state accordingly, and tell all the connected clients.

If we were to allow simply "reloading" the database, avoiding the need to be informed immediately, we would also run into trouble. If the server doesn't know that the underlying database has changed, it could overwrite something the user has written to it. It could also, potentially, run into consistency issues.

So, it would only make sense to be able to reload the database if no clients at all are connected. In which case a restart of the daemon would work just as well.

I'd also like to add that anything you can change in the database directly, you can change via RPC.
If you find anything that is missing, feel free to open a feature request for it.

**Comment by mkrautz** [comment]
Closing, as the DB is not considered part of any public Mumble programming interface.

**Comment by adler187** [comment]
With the advent of Let's Encrypt and dns-01 validation, it would be nice to be able to reload the automatically renewed certificate without have to restart the server.

**Comment by mkrautz** [comment]
Coming in https://github.com/mumble-voip/mumble/pull/2221

**Comment by adler187** [comment]
Not as easy as SIGHUP, but I'll take it. :)
